### PR TITLE
Added nearest interp v2 BF16 FWD kernel

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
New features

### PR changes
OPs

### Describe
Added nearest interp v2 BF16 FWD kernel, fixed some tests and enabled NHWC support

### CODE REVIEW

File: paddle/fluid/operators/interpolate_op.cc [hunk 1]
```diff
@@ -343,21 +343,6 @@ class InterpolateOp : public framework::OperatorWithKernel {
   framework::OpKernelType GetKernelTypeForVar(
       const std::string& var_name, const Tensor& tensor,
       const framework::OpKernelType& expected_kernel_type) const override {
-#ifdef PADDLE_WITH_MKLDNN
```
Comment (jczaja) [inline paddle/fluid/operators/interpolate_op.cc]: why this was removed?

File: paddle/fluid/operators/interpolate_v2_op.cc [hunk 2]
```diff
@@ -429,21 +429,6 @@ class InterpolateV2Op : public framework::OperatorWithKernel {
   framework::OpKernelType GetKernelTypeForVar(
       const std::string& var_name, const Tensor& tensor,
       const framework::OpKernelType& expected_kernel_type) const override {
-#ifdef PADDLE_WITH_MKLDNN
```
Comment (jczaja) [inline paddle/fluid/operators/interpolate_v2_op.cc]: why this was removed?

File: paddle/fluid/operators/interpolate_op.cc (hunk 1 above)
Comment (jakpiase) [reply paddle/fluid/operators/interpolate_op.cc]: discussed offline, I have fixed that

File: paddle/fluid/operators/interpolate_v2_op.cc (hunk 2 above)
Comment (jakpiase) [reply paddle/fluid/operators/interpolate_v2_op.cc]: discussed offline, I have fixed that

**Comment by jakpiase** [comment]
@Silv3S please review this PR

**Comment by lidanqing-vv** [comment]
@baoachun Hi, Please review this PR

**Comment by Silv3S** [comment]
LGTM

**Review by baoachun — APPROVED** [review]
LGTM

**Review by lidanqing-vv — APPROVED** [review]
LGTM

**Pull request merged**